source/common/http/conn_pool_grid.cc

```diff
@@ -22,52 +22,77 @@ ConnectivityGrid::WrapperCallbacks::WrapperCallbacks(ConnectivityGrid& grid,
                                                      Http::ResponseDecoder& decoder,
                                                      PoolIterator pool_it,
                                                      ConnectionPool::Callbacks& callbacks)
-    : grid_(grid), decoder_(decoder), inner_callbacks_(callbacks), pool_it_(pool_it) {
-  // TODO(#15649) add trace logging.
-  cancellable_ = pool().newStream(decoder_, *this);
+    : grid_(grid), decoder_(decoder), inner_callbacks_(callbacks),
+      failover_timer_(grid_.dispatcher_.createTimer([this]() -> void { tryAnotherConnection(); })) {
+  current_ = pool_it;
+  auto state = std::make_unique<ConnectionAttemptState>(*this, pool_it);
+  LinkedList::moveIntoList(std::move(state), connection_attempts_);
+  // TODO(#15649) When adding config for the grid, make this configurable.
+  failover_timer_->enableTimer(std::chrono::milliseconds(5));
```

Comment:
I'd suggest a default of 300ms? 5ms won't give the first one a chance to succeed before trying the second.